python/src/wallets/crossmint/goat_wallets/crossmint/solana_smart_wallet.py

```
@@ -0,0 +1,251 @@
+from typing import Dict, List, Optional, Any, cast
+import base58
+import time
+from solders.instruction import Instruction
+from solders.pubkey import Pubkey
+from solders.message import Message
+from solana.rpc.api import Client as SolanaClient
+from goat.classes.wallet_client_base import Balance, Signature
+from goat_wallets.solana import SolanaWalletClient, SolanaTransaction
+from .api_client import CrossmintWalletsAPI
+from .parameters import SolanaSmartWalletTransactionParams, DelegatedSignerPermission
+from .base_wallet import BaseWalletClient, get_locator
+
+
+class SolanaSmartWalletClient(SolanaWalletClient, BaseWalletClient):
+    def __init__(
+        self,
+        address: str,
+        api_client: CrossmintWalletsAPI,
+        connection: SolanaClient,
+        options: Dict
+    ):
+        SolanaWalletClient.__init__(self, connection)
+        BaseWalletClient.__init__(self, address, api_client, "solana")
+        self.connection = connection
+        self._locator = get_locator(address, None, "solana-smart-wallet")
+
+    def get_address(self) -> str:
+        return self._address
+
+    def sign_message(
```

This method should throw unsupported!

---

Fixed!